core/src/main/java/org/mskcc/cbio/portal/servlet/GetCoExpressionJSON.java

```diff
@@ -108,143 +120,158 @@ protected void doPost(HttpServletRequest httpServletRequest,
-                            Double[] _new_query_gene_exp = new_query_gene_exp_arrlist.toArray(new Double[0]);
-                            Double[] _new_compared_gene_exp = new_compared_gene_exp_arrlist.toArray(new Double[0]);
-                            //convert double object to primitive data
-                            double[] new_query_gene_exp = new double[_new_query_gene_exp.length];
-                            double[] new_compared_gene_exp = new double[_new_compared_gene_exp.length];
-                            for (int m = 0; m < _new_query_gene_exp.length; m++) {
-                                new_query_gene_exp[m] = _new_query_gene_exp[m].doubleValue();
-                                new_compared_gene_exp[m] = _new_compared_gene_exp[m].doubleValue();
-                            }
-                            if (new_query_gene_exp.length != 0 && new_compared_gene_exp.length != 0 &&
-                                compared_gene_id != queryGeneId) {
-                                double pearson = pearsonsCorrelation.correlation(new_query_gene_exp, new_compared_gene_exp);
-                                double spearman = spearmansCorrelation.correlation(new_query_gene_exp, new_compared_gene_exp);
-                                CanonicalGene comparedGene = daoGeneOptimized.getGene(compared_gene_id);
-                                fullResutlStr.append(
-                                    comparedGene.getHugoGeneSymbolAllCaps() + "\t" +
-                                    comparedGene.getCytoband() + "\t" +
-                                    (double) Math.round(pearson * 100) / 100 + "\t" +
-                                    (double) Math.round(spearman * 100) / 100 + "\n"
-                                );
-                            }
-                        }
-                    }
-                    //construct file name
-                    String fileName = "coexpression_" + geneSymbol + "_" +
-                        final_gp.getProfileName().replaceAll("\\s+", "_") + "_" +
-                        cancerStudyIdentifier.replaceAll("\\s+", "_") + ".txt";
-
-                    httpServletResponse.setContentType("text/html");
-                    httpServletResponse.setContentType("application/force-download");
-                    httpServletResponse.setHeader("content-disposition", "inline; filename='" + fileName + "'");
-                    PrintWriter out = httpServletResponse.getWriter();
-                    JSONValue.writeJSONString(fullResutlStr, out);
-                } catch (DaoException e) {
-                    System.out.println(e.getMessage());
-                }
-            } else {
-                JSONObject emptyResult = new JSONObject();
-                httpServletResponse.setContentType("application/json");
-                PrintWriter out = httpServletResponse.getWriter();
-                JSONValue.writeJSONString(emptyResult, out);
-            }
-        }
+		try {
```

Comment:
Although this is easier to follow than EnrichmentsJSON.doPost(), is it safer to just check and punt at the top instead of trying to wrap this entire routine around the accessControl check?